.github/workflows/build_monolithic.yml

```diff
@@ -181,152 +203,113 @@ jobs:
+          winearch: "wine64"
+          build-dir: "${{ env.BUILDROOT }}/wine64"
+          crossflags: "-g -O2 -fcommon"
+          configure-params: "--enable-win64 --with-vulkan"
 
       ############ Build 64bit Version ##############
 
-      - name: Configure wine64
-        env:
-          CC: clang
-          CXX: clang++
-          # see https://github.com/Gcenx/macOS_Wine_builds/issues/17#issuecomment-750346843
-          CROSSCFLAGS: "-g -O2 -fcommon"
-          # Xcode12 by default enables '-Werror,-Wimplicit-function-declaration' (49917738)
-          # this causes wine(64) builds to fail so needs to be disabled.
-          # https://developer.apple.com/documentation/xcode-release-notes/xcode-12-release-notes
-          CFLAGS: "-g -O2 -Wno-implicit-function-declaration -Wno-deprecated-declarations -Wno-format"
-          LDFLAGS: "-Wl,-headerpad_max_install_names"
-        run: |
-          set -x
-
-          export SDL2_CFLAGS="-I$(brew --prefix sdl2)/include"
-          export GPHOTO2_CFLAGS="-I$(brew --prefix libgphoto2)/include -I$(brew --prefix libgphoto2)/include/gphoto2"
-          export GPHOTO2_PORT_CFLAGS="-I$(brew --prefix libgphoto2)/include -I$(brew --prefix libgphoto2)/include/gphoto2"
-
-          export PNG_CFLAGS="-I$(brew --prefix libpng)/include"
-          export PNG_LIBS="-L$(brew --prefix libpng)/lib"
-
-          export LDFLAGS="-L $(brew --prefix molten-vk)/lib"
-
-          mkdir -p ${{ env.BUILDROOT }}/wine64
-          pushd ${{ env.BUILDROOT }}/wine64
-          ${{ env.WINE_CONFIGURE }} \
-                    --enable-win64 \
-                    --disable-tests \
-                    --without-alsa \
-                    --without-capi \
-                    --without-dbus \
-                    --without-inotify \
-                    --without-oss \
-                    --without-pulse \
-                    --without-udev \
-                    --without-v4l2 \
-                    --without-gsm \
-                    --with-png \
-                    --with-sdl \
-                    --with-vulkan \
-                    --without-x
-          popd
-
       - name: Build wine64
-        run: |
-          pushd ${{ env.BUILDROOT }}/wine64
-          make -j$(sysctl -n hw.ncpu 2>/dev/null)
-          popd
+        uses: ./.github/actions/make
+        with:
+          build-dir: "${{ env.BUILDROOT }}/wine64"
 
+      ############ Configure 32on64bit Version ##############
 
-      ############ Build 32bit Version (WoW64) ##############
+      - name: Configure wine32on64 (specific to 21)
+        if: startsWith(matrix.CROSS_OVER_VERSION, '21')
+        uses: ./.github/actions/configure_wine
+        with:
+          winearch: "wine32on64"
+          build-dir: "${{ env.BUILDROOT }}/wine32on64"
+          crossflags: "-g -O2"
+          configure-params: "--enable-win32on64 --with-wine64=${{ env.BUILDROOT }}/wine64 --without-cms --without-gstreamer --without-gphoto --without-sane --without-krb5"
```

Comment:
Still require the Vulkan disables for CX21, maybe the CX20/21 configure options squashed together.

Reply:
This I don't understand. If you look at https://github.com/GabLeRoux/macos-crossover-cloud-build/runs/3257206753 , win32on64 was configured without all the disable-vk-whatever and it went through...

Reply:
If you check the last run you’ll see;

```
fatal error: too many errors emitted, stopping now [-ferror-limit=]
36781
20 errors generated.
36782
make: *** [dlls/winemac.drv/vulkan.o] Error 1
36783
make: *** Waiting for unfinished jobs....
36784
Error: Process completed with exit code 2.
```

I have a feeling the prior wine32on64 stages winemac.drv was somehow getting skipped, this would match my findings with winebuild my my local system.

Reply:
Checked the configure log of when wine32on64 completed and I now know why it completed

```
configure: libvulkan and libMoltenVK 64-bit development files not found, Vulkan won't be supported.
```

Prior `export LDFLAGS="-L $(brew --prefix molten-vk)/lib"` wasn't getting passed to wine32on64 configure only for wine64 configure, that solves that annoyance. And yes wine32on64 still doesn't have Vulkan trunking we lightly won't see this until upstream 32to64 implementation is completed.

Reply:
Oh, this makes things clearer. So I should have added "--with-whatever" instead of just removing "--without-whatever" to ensure it fails if the "whatever" component is not available...

Reply:
To override any configure options already set you can just add --without- later on.

For winebuild I have a text file that contains all desired configure options and wine32on64 appends the needed --without- options to save myself headaches, also doing the same for wine (32Bit) to disable Vulkan/vkd3d etc avoiding build errors.